recipe/meta.yaml

```diff
@@ -29,40 +26,32 @@ build:
 
 requirements:
   build:
-    - python                                 # [build_platform != target_platform]
-    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
-    - numpy                                  # [build_platform != target_platform]
-    - {{ compiler('cxx') }}  # [not win]
-    - {{ stdlib('c') }}
-    - clang                  # [win]
-    - clangxx                # [win]
   host:
-    - python
+    - python {{ python_min }}
     - pip
     - setuptools
-    - numpy
   run:
-    # To ensure ABI compatibility, we install the same C++
-    # compiler that was used for building.
-    - {{ cxx_compiler }}_{{ target_platform }}  # [unix]
-    - clang                  # [win]
-    - clangxx                # [win]
-    - python
-    - {{ pin_compatible('numpy') }}
+    - gxx                    # [linux]
+    - clangxx                # [osx or win]
+    - python >={{ python_min }}
     - decorator
     - gast 0.5.*
     - ply >=3.4
     - colorlog
     - beniget 0.4.*
+    - numpy
+    - __linux                # [linux]
+    - __osx                  # [osx]
+    - __win                  # [win]
 
 test:
   requires:
-    - setuptools  # [py>=312]
+    - setuptools <=73.0.1
```

Comment:
```suggestion
    # fix an issue with numpy.distutils and python 3.9 setuptools
    - setuptools <=73.0.1
```

Reply:
"an issue" is still too vague to actually go check in a few weeks/months whether the cap is still necessary. Please provide a reference

Reply:
Looking around, this looks related to https://github.com/numpy/numpy/issues/27405; however, we're not getting backports of https://github.com/numpy/numpy/pull/27406 because numpy 2.1 dropped python 3.9.

@rgommers, I think it would make sense to backport that patch to numpy 2.0.x, WDYT?

PS. This took 3min of googling. A stacktrace is hardly a replacement as documentation, because it just leaves the task of cleaning this up for later.

Reply:
> I think it would make sense to backport that patch to numpy 2.0.x

https://github.com/conda-forge/numpy-feedstock/pull/337